### PR TITLE
API improvements and examples code review

### PR DESCRIPTION
**Pull request by paolopal**
API is improved to support all fonts and Auto Scroll feature is added, see new web_api.py example.
Examples are reworked to have by default no text rotation (assumed it is scrollphathd library, no scrollbot one).

### CODE REVIEW

**Comment by Gadgetoid** [comment]
Thank you for this! I haven't had a proper look through it yet, since I don't have a Pi  or Scroll pHAT HD to hand for testing, but I'll check over it tomorrow morning. This sort of in-depth PR and code review is really appreciated.

File: library/scrollphathd/api/http.py [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 
 try:
     from queue import Queue
 except ImportError:
     from Queue import Queue
 
+try:
+    from queue import Empty
+except ImportError:
+    from Queue import Empty
```
[the remark this answers is not in the document]
Comment (paolopal) [reply library/scrollphathd/api/http.py]: It's first time for me changing/writing python code (few hours in a couple of weekend) :-)
For sure the merge you proposed is perfect :-)

File: library/scrollphathd/api/http.py [hunk 2]
```diff
@@ -91,21 +109,33 @@ def flip():
...
+            if action.action_type == "flip":
+                scrollphathd.flip(x=action.data[0], y=action.data[1])
+        except Empty:
+            pass
 
         scrollphathd.show()
 
+        if auto_scroll is True:
+            scrollphathd.scroll()
+            time.sleep(0.1)
```
[the remark this answers is not in the document]
Comment (paolopal) [reply library/scrollphathd/api/http.py]: You are right, sorry for poor handling of auto scrolling. If it is ok for you, I'll try for something better in next weekend. :-)

[the remark this answers is not in the document]
Comment (paolopal) [reply library/scrollphathd/api/http.py]: In main function there is this code:
    # To check if it is visible!
    scrollphathd.set_brightness(0.1)

    scrollphathd.write_string(str(args.port), x=1, y=1)
    scrollphathd.show()
It should show port number on display but it seems not working, I had no time to investigate/understand what is expected to do...

**Comment by paolopal** [comment]
Hi Phil,
first of all thanks a Iot giving me the opportunity to have a constructive and helpful debate. I don't have too much time and python is quite new for me! I reworked the code and I hope I improved it.
Waiting for comments,
BR,
\Paolo

**Comment by Gadgetoid** [comment]
Thank you for spending the time to make this code better and clearer for everyone! I love what you've done with the web API. It's very much appreciated!

**Comment by paolopal** [comment]
Hi Phil,
thanks, thanks a lot for appreciation :-) started working on numpy removal from the driver, here it is important to take care about performances, so interesting stuff to go deeper in python and check what happens. I also started creating nodes for node-red, one step at a time.
BR,
\Paolo

**Pull request merged**